apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx

```diff
@@ -629,34 +631,35 @@ export const ScriptBasedFlow: FunctionComponent<AdaptiveScriptsPropsInterface> =
                 }
             >
                 <Dropdown.Menu>
-                    <Input
-                        data-testid={ `${ testId }-secret-search` }
-                        icon="search"
-                        iconPosition="left"
-                        className="search"
-                        placeholder= { t("console:develop.features.applications.edit.sections.signOnMethod" +
-                                        ".sections.authenticationFlow.sections.scriptBased.secretsList.search") }
-                        onChange={ ( data:ChangeEvent<HTMLInputElement>
-                        ) => {
-                            if(!data.currentTarget.value) {
-                                setFilteredSecretList(secretList);
-                            } else {
-                                setFilteredSecretList(secretList.
-                                filter((secret: SecretModel) => secret.secretName.includes(
-                                    data.currentTarget.value)));
-                            }
-                        } }
-                        onClick={ e => e.stopPropagation() }
-                    />
-
+                    {
+                        secretList.length > 0 && (
+                            <Input
+                                data-testid={ `${testId}-secret-search` }
+                                icon="search"
+                                iconPosition="left"
+                                className="search"
+                                placeholder={ t("console:develop.features.applications.edit.sections.signOnMethod" +
+                                    ".sections.authenticationFlow.sections.scriptBased.secretsList.search") }
+                                onChange={ (data: ChangeEvent<HTMLInputElement>
+                                ) => {
+                                    if (!data.currentTarget.value) {
```

Comment:
Shall we add some null checks for safety.
```suggestion
                                    if (!data.currentTarget?.value) {
```